### ISSUE TITLE
[Question] ES5

### ISSUE THREAD

**Issue by mkbctrl** [d4rekanguok/gatsby-typescript#88]
Hey, since in the docs you ask to set at least ES2015 as a target, can you please tell me what will happen if I will use ES5? Can't find any information about what can break.

Thanks in advance!

**Comment by ricokahler** [comment]
This plugin actually combines with the default "JS loader" that Gatsby uses. If you're unfamiliar, Gatsby uses webpack behind the scenes and we simply add the [`ts-loader`](https://github.com/TypeStrong/ts-loader) in the chain before Gatsby's default JS loader.

https://github.com/d4rekanguok/gatsby-typescript/blob/69a7240486f06f88837ab58a7991e0b64a918ce2/packages/gatsby-plugin-ts/src/gatsby-node.ts#L46-L58

So the way this works is:

1. webpack loads a resource and tries to find matching rule for the file extension (in our case, it's `.ts` or `.tsx` files)
2. webpack finds our rule for `.ts`/`.tsx` files and runs our loader chain.
3. in our loader chain, first the typescript loader is run which compiles typescript with the typescript compiler (via the ts-loader). In this step we don't actually want to compile es2015 to es5, we just want to compile typescript-only features like `const enum`s and `namespace`s so that the next gatsby-provided loader can accept our input (e.g. `ts` => `js`)
4. by default, the gatsby-provided loader is the `babel-loader` which uses babel to compile es2015 to whatever target your specify (e.g. `ES2015` => `ES5`)

---

TL;DR you actually want the target to remain as ES2015 for typescript because we feed that result into Gatsby's default JS loader which handles ES2015 to ES5.

If you're looking to support older browsers, there's actually a much simpler way to do so via `browerslist`.

[Check out the official Gatsby docs for more info.](https://www.gatsbyjs.org/docs/browser-support/#specify-what-browsers-your-project-supports-using-browserslist)

**Comment by d4rekanguok** [comment]
Thanks for clarifying Rico, I'll add this to the docs.
